src/TestConsoleApp/RefitExampleCalls.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.Buffers.Text;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using TestApp.Models;
+using TestApp.Models.StarWarsAPI;
+using TestApp.Services;
+using Xablu.WebApiClient;
+
+namespace TestConsoleApp
+{
+    public class RefitExampleCalls
+    {
+        static List<Starships> items = new List<Starships>();
+
+        // 2 ways to check an object. One from json printed, the other an object from the api
+        // the first way is more modular and works for any json, the second more adjusted for SWAPI
+        public static async Task<IEnumerable<Starships>> GetStarShipItemsAsync(bool forceRefresh = false)
+        {
+            IWebApiClient<IStarwarsApi> _webApiClient = WebApiClientFactory.Get<IStarwarsApi>("https://swapi.co/api", false, () => new SampleHttpClientHandler());
```

Comment:
Not particularly wrong or anything like that, but is there any reason why you are passing along the `SampleHttpClientHandler` here?

Reply:
This is the way the calls are being used with your implementation in the testapp. I used those as a guideline!

Reply:
I tried implementing a new DelegatingHandler but that didn't work. Also i have no clue what this is even doing...

Reply:
Well, in that case I would suggest you search around for what `DelegatingHandler` is and what does it do... I can recommend you this [blog post](https://www.stevejgordon.co.uk/httpclientfactory-aspnetcore-outgoing-request-middleware-pipeline-delegatinghandlers) (just ignore the section "Registering Handlers" because it's for web dev).

Also debugging and setting breakpoints can help understand how does the flow go - and parameter names. The second parameter is what you want to have set to `true` instead of changing the bool value in the `DisableAutoRedirects` method, and the third parameter in this case is unnecessary